python/pyarrow/tests/test_cython.py

```diff
@@ -123,18 +123,31 @@ def test_cython_api(tmpdir):
         # pyarrow imported first.
         code = """if 1:
             import sys
+            import os
+
+            try:
+                # Add dll directory was added on python 3.8
+                # and is required in order to find extra DLLs
+                # only for win32
+                for dir in {library_dirs}:
+                    os.add_dll_directory(dir)
+            except AttributeError:
+                pass
 
             mod = __import__({mod_name!r})
             arr = mod.make_null_array(5)
             assert mod.get_array_length(arr) == 5
             assert arr.null_count == 5
-        """.format(mod_name='pyarrow_cython_example')
+        """.format(mod_name='pyarrow_cython_example',
+                   library_dirs=pa.get_library_dirs())
 
-        if sys.platform == 'win32':
+        if sys.platform == 'win32' and not \
+                getattr(os, 'add_dll_directory', False):
+            # Python 3.8 onwards don't check extension module DLLs on path
+            # we have to use os.add_dll_directory instead.
             delim, var = ';', 'PATH'
```

Comment:
```suggestion
        if sys.platform == 'win32':
            if not getattr(os, 'add_dll_directory', False):
                # Python 3.8 onwards don't check extension module DLLs on path
                # we have to use os.add_dll_directory instead.
                delim, var = ';', 'PATH'
```

Reply:
Also the `getattr` check can simply be `hasattr(os, 'add_dll_directory')`